jwst/pipeline/calwebb_image3.py

```diff
@@ -41,77 +40,63 @@ class Image3Pipeline(Pipeline):
                  'source_catalog': source_catalog_step.SourceCatalogStep
                  }
 
-    def process(self, input):
+    def process(self, input_data):
         """
         Run the Image3Pipeline
 
         Parameters
         ----------
-        input: Level3 Association, or ModelContainer
+        input_data: Level3 Association, or ModelContainer
             The exposures to process
         """
-
         self.log.info('Starting calwebb_image3 ...')
 
         # Only load science and background members from input ASN
         asn_exptypes = ['science', 'background']
-        input_models = datamodels.open(input, asn_exptypes=asn_exptypes)
-
-        # If input is an association, set the output to the product name.
-        try:
-            self.output_file = input_models.meta.asn_table.products[0].name
-        except AttributeError:
-            pass
-
-        # Check if input is single or multiple exposures
-        is_container = isinstance(input_models, datamodels.ModelContainer)
-        try:
-            has_groups = len(input_models.group_names) > 1
-        except Exception:
-            has_groups = False
-        if is_container and has_groups:
-            if is_moving_target(input_models):
-                self.log.info("Assigning WCS to a Moving Target exposure.")
-                input_models = self.assign_mtwcs(input_models)
-            else:
-                self.log.info("Aligning input images...")
-                input_models = self.tweakreg(input_models)
-
-            self.log.info("Matching sky values across all input images...")
-            input_models = self.skymatch(input_models)
-
-            self.log.info("Performing outlier detection on input images...")
-            input_models = self.outlier_detection(input_models)
-
-            if input_models[0].meta.cal_step.outlier_detection == 'COMPLETE':
-                self.log.info("Writing Level 2c images with updated DQ arrays...")
-                # Set up Level 2c suffix to be used later
-                asn_id = input_models.meta.asn_table.asn_id
-                suffix_2c = '{}_{}'.format(asn_id, 'crf')
-                for model in input_models:
-                    self.save_model(
-                        model,
-                        output_file=model.meta.filename,
-                        suffix=suffix_2c
-                    )
-
-        self.log.info("Resampling images to final result...")
-        result = self.resample(input_models)
-
-        try:
-            result.meta.asn.pool_name = input_models.meta.asn_table.asn_pool
-            result.meta.asn.table_name = os.path.basename(input)
-            result.meta.filename = input_models.meta.asn_table.products[0].name
-        except Exception:
-            pass
```

Comment:
I noticed that some of the other pipelines (for example `Spec3Pipeline`) don't include this section, so I wasn't sure if it was necessary or not.

Reply:
I think at least some of that might be included down in the `resample` step itself when the output data model (`result`) is created and has `.update()` called to copy meta from one of the inputs. I guess the regtests will tell us for sure (i.e. we'll see if ASNPOOL and ASNTABLE are now missing from the outputs). And I'm pretty sure that `result.meta.filename` is set down within the `datamodels` code itself when any file gets saved to disk.

Reply:
Removed this section and it resulted in no changes to ASNPOOL or ASNTABLE in regression tests